models/appModel.js

```diff
@@ -0,0 +1,61 @@
+const db = require("../db/connection");
+const fs = require("fs/promises")
+const format = require("pg-format")
+
+exports.fetchAllTopics = () => {
+    return db
+        .query(`SELECT * FROM topics;`)
+        .then((topics) => topics.rows);
+}
+
+exports.fetchAllEndpoints = async () => {
+    const data = await fs.readFile("./endpoints.json");
+    const endpoints = JSON.parse(data)
+    return endpoints;
+}
+
+exports.fetchAllArticles = (sort_by = "created_at", order = "DESC") => {
+    return db
+        .query(`
+    SELECT 
+        articles.author, 
+        articles.title, 
+        articles.article_id, 
+        articles.topic, 
+        articles.created_at, 
+        articles.votes, 
+        articles.article_img_url,
+    COUNT(comments.comment_id)::INT AS comment_count
+    FROM 
+        articles 
+    LEFT JOIN 
+        comments
+    ON 
+        articles.article_id = comments.article_id
+    GROUP BY 
+        articles.article_id
+    ORDER BY ${sort_by} ${order};`)
+        .then((articles) => articles.rows)
+
+}
```

Comment:
Great implementation of this model so far, but there's a risk of sql injection for `sort_by` and `order`!

We can't use a parametrized query as that's only for values, not for SQL keywords. So we'd want to keep track of what we'd consider valid sort_by & order values - rejecting the promise with a custom error handler if an invalid one has been passed in. Give this a shot!